File: Day8/HauntedWasteland.py
# Read the file
with open("Day8/input.txt", "r") as file:
    data = file.readlines()

# The first line is the left right instructions
lr = data[0].strip()

print(lr)

# The rest of the lines are the instructions
data = data[2:]

# Each line has the format AAA = (BBB, CCC)
# Record the values of AAA, BBB, and CCC
net = []
for line in data:
    line = line.strip().split(" ")
    net.append((line[0], line[2][1:4], line[3][0:3]))
    print(net[-1])

# Look for 'AAA' in the net
found = False
target = 'AAA'
step = 0
index = 0

while not found:
    # Look for target in the net
    if target == 'ZZZ':
        print('Reached the end of the net')
        found = True
        break
    for i in range(len(net)):
        if net[i][0] == target:
            # Get if we should read the left or right value
            if lr[index] == 'L':
                target = net[i][1]
            else:
                target = net[i][2]
        
            print(target, lr[index])
            index = (index + 1) % len(lr)
            step += 1
            
            break

print('Number of steps:', step)

File: Day8/HauntedWasteland.cpp
// This is a translation of the HauntedWasteland.py file into C++.

#include <iostream>
#include <string>
#include <fstream>
#include <vector>

using namespace std;

int main(){
    string line;
    ifstream file("input_easy.txt");
    vector<string> data;
    vector<vector<string>> net;

    string lr;

    // Read de file and store the data in a vector
    if (file.is_open()){
        while (getline(file, line)){
            data.push_back(line);
        }
        file.close();
    }
    else cout << "Unable to open file";
    

    lr = data[0];
    cout << "The data is: " << lr << endl;

    // Remove the firs two lines of data
    data.erase(data.begin());
    data.erase(data.begin());

    for (int i = 0; i < data.size(); i++){
        cout << data[i] << endl;
    }

    // The rest of the lines are the instructions










    return 0;
}
